src/main/java/com/mraof/minestuck/client/gui/SummonerScreen.java

```diff
@@ -0,0 +1,114 @@
+import com.mraof.minestuck.tileentity.redstone.SummonerTileEntity;
+import net.minecraft.client.gui.screen.Screen;
+import net.minecraft.client.gui.widget.TextFieldWidget;
+import net.minecraft.client.gui.widget.button.Button;
+import net.minecraft.entity.EntityType;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.text.StringTextComponent;
+import net.minecraftforge.fml.client.gui.widget.ExtendedButton;
+
+import java.util.Optional;
+
+public class SummonerScreen extends Screen
+{
+	private static final ResourceLocation guiBackground = new ResourceLocation("minestuck", "textures/gui/generic_medium.png");
+	
+	private static final int guiWidth = 150;
+	private static final int guiHeight = 98;
+	
+	private final SummonerTileEntity te;
+	private boolean isUntriggerable;
+	private int summonRange;
+	
+	public Button incrementButton;
+	public Button decrementButton;
+	private Button unTriggerableButton;
+	
+	private TextFieldWidget entityTypeTextField;
+	
+	
+	SummonerScreen(SummonerTileEntity te)
+	{
+		super(new StringTextComponent("Summoner"));
+		
+		this.te = te;
+		this.summonRange = te.getSummonRange() > 0 ? te.getSummonRange() : 8; //if its defaulted on creation to 0, set it to the intended default of 8
+		this.isUntriggerable = te.getBlockState().getValue(SummonerBlock.UNTRIGGERABLE);
+	}
+	
+	@Override
+	public void init()
+	{
+		int yOffset = (this.height / 2) - (guiHeight / 2);
+		
+		addButton(incrementButton = new ExtendedButton(this.width / 2 + 20, (height - guiHeight) / 2 + 12, 20, 20, new StringTextComponent("+"), button -> changeRange(1)));
+		addButton(decrementButton = new ExtendedButton(this.width / 2 - 40, (height - guiHeight) / 2 + 12, 20, 20, new StringTextComponent("-"), button -> changeRange(-1)));
+		
+		this.entityTypeTextField = new TextFieldWidget(this.font, this.width / 2 - 60, yOffset + 40, 120, 18, new StringTextComponent("Current Entity Type"));	//TODO Use translation instead, and maybe look at other text fields for what the text should be
+		this.entityTypeTextField.setValue(EntityType.getKey(te.getSummonedEntity()).toString());
+		addButton(entityTypeTextField);
+		
+		if(isUntriggerable)
+			addButton(unTriggerableButton = new ExtendedButton(this.width / 2 - 65, yOffset + 70, 85, 20, new StringTextComponent("UNTRIGGERABLE"), button -> cycleUntriggerable()));
+		else
+			addButton(unTriggerableButton = new ExtendedButton(this.width / 2 - 65, yOffset + 70, 85, 20, new StringTextComponent("TRIGGERABLE"), button -> cycleUntriggerable()));
+		addButton(new ExtendedButton(this.width / 2 + 25, yOffset + 70, 40, 20, new StringTextComponent("DONE"), button -> finish()));
+	}
+	
+	/**
+	 * Attempts to increase or decrease the range at which the entity can be summoned if that value is between the min and max
+	 */
+	private void changeRange(int change)
+	{
+		summonRange = MathHelper.clamp(summonRange + change, 1, 64);
```

Comment:
How about also modifying `this.incrementButton.active` and `this.decrementButton.active` depending on if the summon range is at its maximum or minimum?

Reply:
I think I understand but I am not sure what the benefit is or exactly how it would work

Reply:
Oh wait, you mean visually removing the buttons once it is no longer possible to increment in that direction dont you

Reply:
By setting `button.active` to false, the button will darken and become unclickable. It's a good indicator for the player when the range can't go further in one direction. You could also change `button.visible` to false to visually remove it, but I think that `button.active` is more appropriate here.